examples/fluids/src/setupts.c

```diff
@@ -164,6 +166,10 @@ static PetscErrorCode Surface_Forces_NS(DM dm, Vec G_loc, PetscInt num_walls, co
         const PetscInt           p = points[i];
         const StateConservative *r;
         PetscCall(DMPlexPointLocalRead(dm, p, g, &r));
+        for (PetscInt k = 0; k < num_points; k++) {
+          PetscCall(PetscSectionGetDof(s, points[k], &dof));  // there are 5 dofs
+          // printf("points = %d, num_points = %d\n", points[k], num_points);
+        }
```

Comment:
Let's think about what's happening here. `num_points` is the global number of topological points on the wall and we're already looping over those with `i`. The inner loop we need is over the nodes on point `p`. You could `for (PetscInt node=0; node<dof / 5; node++)` and read `r[node].momentum[j]`.

This could be a bit more defensive by checking `PetscSectionGetFieldComponents(s, 0, &num_comp)` to avoid the magic number 5 and help get it correct when we add turbulence variables.

Reply:
@jedbrown, actually in the loop, sometimes ```dof``` is ```2``` but ```num_comp``` is 5. Please, could you explain why you did this ```dof / 5``` ?

Reply:
I don't think it should be. There are only 2 global dofs, but the local section should say there are 5 dofs, 3 of which are constrained (or 10 dofs, 6 of which are constrained). Can you give precise instructions to reproduce that `dof = 2`?

Reply:
Yeah it says 5 dofs but for example if I print ```dof``` before line
```for (PetscInt node = 0; node < dof / num_comp; node++)```
then, it says 2 and 5 (but mostly 5).